source/assets/js/components/fef-generic-expander.js

```diff
@@ -0,0 +1,138 @@
+    constructor(element) {
+        this.$element = $(element);
+        this.openToggleClass = this.$element.attr('data-genex-open-toggle-class') || 'js-genex-toggle-open';
+        this.openPanelClass  = this.$element.attr('data-genex-open-panel-class')  || 'js-genex-panel-open';
+
+        this.eventSource = this.$element.attr('data-event-source');
+        this.eventValue = this.$element.attr('data-event-value');
+
+        this.initA11Y();
+        this.bindEvents();
+
+        this.isTogglingAllowed = true;
+    }
+
+    initA11Y() {
+        // set aria-expand attribute
+        $(TOGGLE_CLASS).attr('aria-expanded', false);
+
+        // set aria-controls attribute
+        $('[data-genex-target-id]').each((index, element) => {
+            $(element).attr('aria-controls', $(element).attr('data-genex-target-id'));
+        });
+    }
+
+    bindEvents() {
+        let clickHandler = (event) => {
+            event.preventDefault();
+            this.togglePanels(event);
+        };
+
+        let keyboardHandler = (event) => {
+            if (event.keyCode === KEYCODES.enter || event.keyCode === KEYCODES.space) {
+                event.preventDefault();
+                this.togglePanels(event);
+            }
+        };
+
+        this.$element.on('click', TOGGLE_CLASS, clickHandler);
+        this.$element.on('keydown', TOGGLE_CLASS, keyboardHandler);
+    }
+
+    togglePanels(event) {
+        //prevent toggling of multiple panels
+        if (this.isTogglingAllowed) {
+            let isSwitching = false;
+
+            const $lastToggle = $('.' + this.openToggleClass, this.$element);
+            const $currentToggle = $(event.currentTarget);
+            const $openPanel = $('.' + this.openPanelClass, this.$element);
+
+            if ($openPanel.length > 0) {
+                if ($lastToggle.get(0) !== $currentToggle.get(0)) {
+                    isSwitching = true;
+                    this.closeOpenPanels(event, $lastToggle, $openPanel, () => {
+                        this.openCurrentPanel(event, $currentToggle);
+                    });
+                } else {
+                    this.closeOpenPanels(event, $lastToggle, $openPanel);
+                }
+            } else {
+                this.openCurrentPanel(event, $currentToggle);
+            }
+            //blocking toggling for duration of animation
+            //if panels are switched 2 animations are played consecutively, so the duration is doubled
+            this.isTogglingAllowed = false;
+            setTimeout(() => {this.isTogglingAllowed = true;}, !isSwitching ? ANIMATION_DEFAULT_DURATION : ANIMATION_DEFAULT_DURATION * 2);
+        }
+    }
+
+    closeOpenPanels(event, $lastToggle, $openPanel, callbackFunction) {
+        this.setA11YState($(TOGGLE_CLASS), false);
+        this.doTracking(event, false);
+
+        $lastToggle.removeClass(this.openToggleClass);
+
+        $openPanel.slideUp(ANIMATION_DEFAULT_DURATION, ANIMATION_DEFAULT_EASING, () => {
```

Comment:
```suggestion
        $openPanel.slideUp(
                ANIMATION_DEFAULT_DURATION,
                ANIMATION_DEFAULT_EASING,
                () => {
```

(gleich wie bei `openCurrentPanel`)